libprecompiled/extension/PaillierPrecompiled.cpp

```diff
@@ -0,0 +1,79 @@
+/*
+ * @CopyRight:
+ * FISCO-BCOS is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * FISCO-BCOS is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with FISCO-BCOS.  If not, see <http://www.gnu.org/licenses/>
+ * (c) 2016-2018 fisco-dev contributors.
+ */
+/** @file PaillierPrecompiled.h
+ *  @author shawnhe
+ *  @date 20190808
+ */
+
+#include "PaillierPrecompiled.h"
+#include <libdevcore/Common.h>
+#include <libethcore/ABI.h>
+#include <paillier/callpaillier.h>
+#include <string>
+
+using namespace dev;
+using namespace dev::blockverifier;
+using namespace dev::precompiled;
+
+const char* const PAILLIER_METHOD_SET_STR = "paillierAdd(string,string)";
+
+PaillierPrecompiled::PaillierPrecompiled()
+{
+    m_callPaillier = std::make_shared<CallPaillier>();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'm_callPaillier' is assigned in constructor body. Consider performing initialization in initialization list.](https://app.codacy.com/gh/FISCO-BCOS/FISCO-BCOS/pullRequest?prid=5173586)